raiden/transfer/secret_registry.py

```diff
@@ -14,16 +20,34 @@ def events_for_onchain_secretreveal(
     secret: Secret,
     expiration: BlockExpiration,
     block_hash: BlockHash,
+    target_state: TargetTransferState = None,
+    pseudo_random_generator: random.Random = None
 ) -> List[Event]:
     events: List[Event] = list()
 
     if not isinstance(secret, T_Secret):
         raise ValueError("secret must be a Secret instance")
 
     if get_status(channel_state) in CHANNEL_STATES_PRIOR_TO_CLOSED + (CHANNEL_STATE_CLOSED,):
-        reveal_event = ContractSendSecretReveal(
-            expiration=expiration, secret=secret, triggered_by_block_hash=block_hash
-        )
-        events.append(reveal_event)
+        if not channel_state.is_light_channel:
+            return [
+                ContractSendSecretReveal(
+                    expiration=expiration,
+                    secret=secret,
+                    triggered_by_block_hash=block_hash,
+                )
+            ]
+        elif target_state is not None:
+            return [
+                StoreMessageEvent(
+                    message_id=message_identifier_from_prng(pseudo_random_generator),
+                    payment_id=target_state.transfer.payment_identifier,
+                    light_client_address=channel_state.our_state.address,
+                    message_order=0,
+                    message=RequestRegisterSecret(),
+                    is_signed=False,
+                    message_type=LightClientProtocolMessageType.RequestRegisterSecret
+                )
+            ]
 
-    return events
+    return []
```

Comment:
can you explain this change? it also makes the default assignment of `events` (at the signature level) useless